Sources/generate-swift-syntax-builder/Templates/TokenFile.swift

```diff
@@ -0,0 +1,288 @@
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+let tokenFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+
+  let tokenType = SyntaxBuildableType(syntaxKind: "Token")
+  let conformances = ["SyntaxBuildable"] + tokenType.convertibleToTypes.map(\.expressibleAsBaseName)
+  + tokenType.elementInCollections.map(\.expressibleAsBaseName)
+
+
+  StructDecl(
+    leadingTrivia: .newline +
+      .docLineComment("""
+      /// Represents `TokenSyntax` in `SwiftSyntaxBuilder`.
+      /// At the moment, this just wraps `TokenSyntax`, but we can make it store just the information necessary to build a `TokenSyntax` in the future.
+      """) +
+      .newline,
+    modifiers: [Token.public],
+    identifier: "Token",
+    inheritanceClause: createTypeInheritanceClause(conformances: conformances)
+  ) {
+    VariableDecl(.let, name: "tokenSyntax", type: "TokenSyntax")
+
+    VariableDecl(
+      leadingTrivia: .newline,
+      letOrVarKeyword: .var
+    ) {
+      PatternBinding(pattern: "text",
+                     typeAnnotation: "String",
+                     accessor: CodeBlock {
+        MemberAccessExpr(base: "tokenSyntax", name: "text")
+      })
+    }
+
+    InitializerDecl(
+      leadingTrivia: .newline,
+      modifiers: [Token.public],
+      signature: FunctionSignature(
+        input: ParameterClause() {
+          FunctionParameter(
+            firstName: .identifier("tokenSyntax"),
+            colon: .colon,
+            type: "TokenSyntax"
+          )
+        }
+      )
+    ) {
+      SequenceExpr {
+        MemberAccessExpr(base: "self", name: "tokenSyntax")
+        AssignmentExpr()
+        IdentifierExpr("tokenSyntax")
+      }
+    }
+
+    for leadingTrailing in ["Leading", "Trailing"] {
+      FunctionDecl(
+        leadingTrivia: .newline,
+        modifiers: [Token.public],
+        identifier: .identifier("with\(leadingTrailing)Trivia"),
+        signature: FunctionSignature(
+          input: ParameterClause {
+            FunctionParameter(
+              firstName: .wildcard,
+              secondName: .identifier("trivia"),
+              colon: .colon,
+              type: "Trivia"
+            )
+          },
+          output: "Token"
+        )
+      ) {
+        FunctionCallExpr("Token") {
+          TupleExprElement(
+            label: "tokenSyntax",
+            expression: FunctionCallExpr(MemberAccessExpr(base: "tokenSyntax", name: "with\(leadingTrailing)Trivia")) {
+              TupleExprElement(expression: IdentifierExpr("trivia"))
```

Comment:
I think we can shorten this to

```suggestion
              TupleExprElement(expression: "trivia")
```